packages/syft/src/syft/service/user/user_service.py

```diff
@@ -325,11 +328,43 @@ def admin_verify_key(self) -> Union[SyftVerifyKey, SyftError]:
         except Exception as e:
             return SyftError(message=str(e))
 
+    @service_method(path="user.toggle_registration", name="toggle_signup")
+    def toggle_registration(
+        self, context: AuthedServiceContext, enable: bool
+    ) -> Union[SyftSuccess, SyftError]:
+        """Enable/Disable Registration for Data Scientist or Guest Users."""
+        flags.CAN_REGISTER = enable
+        method = context.node.get_service_method(SettingsService.update)
+        settings = NodeSettingsUpdate(signup_enabled=enable)
+
+        result = method(context=context, settings=settings)
+
+        if result.is_err():
+            return SyftError(message=f"Failed to update settings: {result.err()}")
+
+        message = "enabled" if enable else "disabled"
+        return SyftSuccess(message=f"Registration feature successfully {message}")
+
     def register(
         self, context: NodeServiceContext, new_user: UserCreate
     ) -> Union[Tuple[SyftSuccess, UserPrivateKey], SyftError]:
         """Register new user"""
 
+        request_user_role = (
+            ServiceRole.GUEST
+            if new_user.created_by is None
```

Comment:
Cant we not just use context.credentials here? Does that remove the need to have this new_user.created_by field?

Reply:
Yes you're right, we can just use context.credentials. I will update it.

Reply:
@koenvanderveen I rechecked, we don't have `context.credentials` at this point. This is because, the `context` is a NodeServiceContext object and not AuthedServiceContext
Refer to: https://github.com/OpenMined/PySyft/blob/8b98a07e66d193a7233cd4120681c1da016f4dde/packages/syft/src/syft/node/routes.py#L144

Reply:
I think it might be security issue shubham, without authedservicecontext.
Since the verify key of the domain is public.

I could create UserCreate Object Customly, and include the verify key of the domain node and post it to the register endpoint.

I think it would good to have signature verification in register endpoint

Reply:
I am thinking, on whether to sign the UserCreate Message or pass the signing key itself.

For for the first option:
When we sign the UserCreate Message , 

we could extract the verify key , from it directly.

For the second option:
Even passing the signing key, directly, should be not be  a problem..
Since signing key is a very important primitive in our system.
I  was thinking , maybe we could avoid passing signing key, during exchange.
I am also not fully sure.

Thinking more on it.

Reply:
First I thought of removing /register endpoint, and we could directly call

`domain_client.api.services.user.register()`

Which would directly give us the authedservicecontext.

But I think ,the /register is used by the frontend team for registration in UI.
Not sure , if it would be a good idea to remove that.

Reply:
the current approach is still not secure I believe. Normally we ensure that the signing key passed into the request is paired with the signing key that is used to sign the message on the client side, and therefore getting by verify key is equivalent to getting by signing key (please correct me if I am wrong here). For .register we are not doing this whole verification dance (as we may not have the keys for the user yet). Therefore we should get the user by signing key in order for this to be secure. In the current scenario you could pass a verify key, which is public, of another user

Reply:
Fixed by: https://github.com/OpenMined/PySyft/pull/7715/commits/b0f3e791af7392c7e670d5232637b1211465a2e8